airflow/models/pool.py

```diff
@@ -224,15 +236,25 @@ def occupied_slots(self, session: Session = NEW_SESSION) -> int:
         """
         from airflow.models.taskinstance import TaskInstance  # Avoid circular import
 
+        occupied_states = Pool.get_occupied_states(self.include_deferred)
+
         return int(
             session.scalar(
                 select(func.sum(TaskInstance.pool_slots))
                 .filter(TaskInstance.pool == self.pool)
-                .filter(TaskInstance.state.in_(EXECUTION_STATES))
+                .filter(TaskInstance.state.in_(occupied_states))
             )
             or 0
         )
 
+    @staticmethod
+    def get_occupied_states(include_deferred):
+        if include_deferred:
+            return EXECUTION_STATES | {
+                TaskInstanceState.DEFERRED,
+            }
+        return EXECUTION_STATES
```

Comment:
Why is this not an instance method so we can simply do `pool.get_occupied_states()` without arguments?

Reply:
Right, that's fixed. Static method was a remnant of a different approach.